sdk/synapse/synapse-artifacts/src/artifactsClient.ts

```diff
@@ -43,23 +62,23 @@ export class ArtifactsClient extends ArtifactsClientContext {
     options?: ArtifactsClientOptionalParams
   ) {
     super(credentials, endpoint, options);
-    this.linkedService = new LinkedService(this);
-    this.dataset = new Dataset(this);
-    this.pipeline = new Pipeline(this);
-    this.pipelineRun = new PipelineRun(this);
-    this.trigger = new Trigger(this);
-    this.triggerRun = new TriggerRun(this);
-    this.dataFlow = new DataFlow(this);
-    this.dataFlowDebugSession = new DataFlowDebugSession(this);
-    this.sqlScript = new SqlScript(this);
-    this.sparkJobDefinition = new SparkJobDefinition(this);
-    this.notebook = new Notebook(this);
-    this.workspace = new Workspace(this);
-    this.sqlPools = new SqlPools(this);
-    this.bigDataPools = new BigDataPools(this);
-    this.integrationRuntimes = new IntegrationRuntimes(this);
-    this.library = new Library(this);
-    this.workspaceGitRepoManagement = new WorkspaceGitRepoManagement(this);
+    this.linkedService = new LinkedServiceImpl(this);
+    this.dataset = new DatasetImpl(this);
+    this.pipeline = new PipelineImpl(this);
```

Comment:
@xirzec
Just thought you'd be interested...would this clash horribly with how you plan to expose the corev2 pipelines in clients? (Did we decide on just a field called `pipeline` or a different strategy?